src/applications/MachineShopSimulator.java

```diff
@@ -44,31 +44,30 @@ boolean moveToNextMachine(Job theJob, SimulationResults simulationResults) {
 
     /**
      * change the state of theMachine
-     * 
+     *
      * @return last job run on this machine
      */
     Job changeState(int theMachine) {// Task on theMachine has finished,
                                             // schedule next one.
         Job lastJob;
-        if (machine[theMachine].getActiveJob() == null) {// in idle or change-over
+
+        if (machine[theMachine].isInactive()) {// in idle or change-over
                                                     // state
             lastJob = null;
             // wait over, ready for new job
-            if (machine[theMachine].getJobQ().isEmpty()) // no waiting job
+            if (machine[theMachine].noJobQueued()) // no waiting job
                 eList.setFinishTime(theMachine, largeTime);
             else {// take job off the queue and work on it
-                machine[theMachine].setActiveJob((Job) machine[theMachine].getJobQ()
-                        .remove());
-                machine[theMachine].setTotalWait(machine[theMachine].getTotalWait() + timeNow
-                        - machine[theMachine].getActiveJob().getArrivalTime());
-                machine[theMachine].setNumTasks(machine[theMachine].getNumTasks() + 1);
-                int t = machine[theMachine].getActiveJob().removeNextTask();
+                machine[theMachine].updateActiveJob();
+                machine[theMachine].setTotalWait(timeNow);
+                machine[theMachine].incNumTasks();
+                int t = machine[theMachine].nextTask();
                 eList.setFinishTime(theMachine, timeNow + t);
             }
         } else {// task has just finished on machine[theMachine]
                 // schedule change-over time
             lastJob = machine[theMachine].getActiveJob();
-            machine[theMachine].setActiveJob(null);
+            machine[theMachine].setInactive();
```

Comment:
I really like all the changes in this class, but this one strikes me as particularly elegant. It's a really small change (almost the same number of characters), but it's a much better name, and it avoids passing `null` in as an argument (which is probably a small smell in its own right). Nice job!